pkg/grpc/portergrpc/installation.go

```diff
@@ -0,0 +1,113 @@
+package portergrpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	iGRPC "get.porter.sh/porter/gen/proto/go/porterapis/installation/v1alpha1"
+	"get.porter.sh/porter/pkg/porter"
+	"get.porter.sh/porter/pkg/tracing"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+func makeInstOptsLabels(labels map[string]string) []string {
+	var retLabels []string
+	for k, v := range labels {
+		retLabels = append(retLabels, fmt.Sprintf("%s=%s", k, v))
+	}
+	return retLabels
+}
+
+func makeGRPCInstallation(inst porter.DisplayInstallation, gInst *iGRPC.Installation) error {
+	bInst, err := json.Marshal(inst)
+	if err != nil {
+		return fmt.Errorf("porter.DisplayInstallation marshal error: %e", err)
+	}
+	pjum := protojson.UnmarshalOptions{}
+	err = pjum.Unmarshal(bInst, gInst)
+	if err != nil {
+		return fmt.Errorf("installation GRPC Installation unmarshal error: %e", err)
+	}
+	return nil
+}
+
+func makeGRPCInstallationOutput(dv porter.DisplayValue, gInstOut *iGRPC.PorterValue) error {
+	bInstOut, err := json.Marshal(dv)
+	if err != nil {
+		return fmt.Errorf("PorterValue marshal error: %e", err)
+	}
+	pjum := protojson.UnmarshalOptions{}
+	err = pjum.Unmarshal(bInstOut, gInstOut)
+	if err != nil {
+		return fmt.Errorf("installation GRPC InstallationOutputs unmarshal error: %e", err)
+	}
+	return nil
+}
+
+func (s *PorterServer) ListInstallations(ctx context.Context, req *iGRPC.ListInstallationsRequest) (*iGRPC.ListInstallationsResponse, error) {
+	ctx, log := tracing.StartSpan(ctx)
+	defer log.EndSpan()
+	p, err := GetPorterConnectionFromContext(ctx)
+	// Maybe try to setup a new porter connection instead of erring?
```

Comment:
In what cases would we not be able to get it from the context?

Reply:
this also raises a question - is it worth possibly just passing the connection as its own parameter to this function?

Reply:
^ second this, perhaps we shouldn't get this far if we haven't established a connection